Fix for Navbar overlapping page content when it overflows

jaxcentracom opened the pull request:
I have noticed that when the menu buttons at the top of the page on the navbar overflow to a new line the navbar overlaps the top of the content underneath. It looks to be caused by the navbar having the navbar-fixed-top class which sets it position to fixed. The only difference I noticed with removing the navbar-fixed-top class was the navbar being offset from the top. I have removed the offset and everything looked good. I also removed a br tag that was adding too much of a gap between the navbar and content.

fixes #16719

I am not sure what the original intent was of doing it this way so if someone knows why that might help me find what these changes might break. So far I have navigated to most pages I visit frequently and everything looks good.

This is a picture of the changes. Usually the dashboard edit options would be slightly obscured. It was a lot worse on other pages but this is a production server I am testing on so  I don't really want to show other pages.
![1732231202_grim](https://github.com/user-attachments/assets/5f75e5ec-a92d-488d-b4c8-1d156235d82a)


DO NOT DELETE THE UNDERLYING TEXT

#### Please note

> Please read this information carefully. You can run `./lnms dev:check` to check your code before submitting.

- [x] Have you followed our [code guidelines?](https://docs.librenms.org/Developing/Code-Guidelines/)
- [x] If my Pull Request does some changes/fixes/enhancements in the WebUI, I have inserted a screenshot of it.
- [x] If my Pull Request makes discovery/polling/yaml changes, I have added/updated [test data](https://docs.librenms.org/Developing/os/Test-Units/).

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/librenms/librenms?pullRequest=16749) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/librenms/librenms?pullRequest=16749) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/librenms/librenms?pullRequest=16749) it.</sub>

laf reviewed (approved):
LGTM.

I think the `<br>` was a legacy fix to keep the body of the site under the navbar many moons ago.

I've tested this and can't see a downside to it. Would like a 2nd opinion from another @librenms/core member.

jaxcentracom commented:
I realized this make make the navbar no longer stick to the top of the window when scrolling. I am not able to test it out right now but I will be able to look into it tomorrow.

laf commented:
Good spot, I'd overlooked that.

murrant commented:
navbar-fixed-top does make it stick to the top of the window, but it also assumes a fixed height of the navbar.  If the navbar goes over that height, you see the issue.

In the past, I've attempted to minimize this by scaling down various elements so that doesn't happen at different screen sizes, but the problem is you can have a variable amount of menu items.  This makes it possible to make the navbar too high with certain screen widths.

jaxcentracom commented:
I managed to make the navbar stick to the page with the other changes I made I just had to add the following CSS attributes to the nav element.

`position: sticky;
 top: 0;
 z-index: 100;`

I am just not sure where it would be best to put these. Should I just put it in the styles.css file?

jaxcentracom commented:
I believe I have it working properly. Also all those commits in the you might see where I force pushed was just me cleaning up accidentally squashing some commits into mine that I shouldn't have. The only things that should be change is the addition of a new navbar-sticky-top CSS class that I added to the navbar and defined in the styles.css file. There is also the br tag I removed from the main layout and the removal of the 50px of padding on top of the body.

laf commented:
I think I need to do some testing on how this looks across various pages in length and content before approving. I'll try and get to that asap.

murrant commented:
I'd like to test this too, it is easy to get this wrong :)

jaxcentracom commented:
I have removed the borders on the sides and  top and I have removed the border radius so the bottom border is flat. I left the bottom border because it provides good distinction between the page content and the navbar.
